services/api_gw.py

```diff
@@ -2,40 +2,181 @@
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources
     )
 
 
 def api_gw_cache_enabled_and_encrypted():
+    apis = v1_client.get_rest_apis()
+    compliant_resources=[]
+    non_compliant_resources=[]
+    if 'items' in apis:
+        for api in apis['items']:
+            stages = v1_client.get_stages(
+                restApiId=api['id'],
+            )
+            if 'item' in stages:
+                non_compliant_resources.extend([
+                    f"arn:aws:apigateway:{v1_client.meta.region_name}::/restapis/{api['id']}/stages/{stage['stageName']}"
+                    for stage in stages['item']
+                    if not stage['methodSettings']
+                ])
+                non_compliant_resources.extend([
+                    f"arn:aws:apigateway:{v1_client.meta.region_name}::/restapis/{api['id']}/stages/{stage['stageName']}"
+                    for stage in stages['item']
+                    for _,settings in stage['methodSettings'].items()
+                    if not settings['cachingEnabled']
+                    or not settings['cacheDataEncrypted']
+                ])
+                compliant_resources.extend([
+                    f"arn:aws:apigateway:{v1_client.meta.region_name}::/restapis/{api['id']}/stages/{stage['stageName']}"
+                    for stage in stages['item']
+                    for _,settings in stage['methodSettings'].items()
+                    if settings['cachingEnabled']
+                    and settings['cacheDataEncrypted']
+                ])
     return RuleCheckResult(
-        passed=False, compliant_resources=[], non_compliant_resources=[]
+        passed=not non_compliant_resources,
+        compliant_resources=compliant_resources,
+        non_compliant_resources=non_compliant_resources
     )
 
 
 def api_gw_execution_logging_enabled():
+    apis = v1_client.get_rest_apis()
+    compliant_resources=[]
+    non_compliant_resources=[]
+    if 'items' in apis:
+        for api in apis['items']:
+            stages = v1_client.get_stages(
+                restApiId=api['id'],
+            )
+            if 'item' in stages:
+                non_compliant_resources.extend([
+                    f"arn:aws:apigateway:{v1_client.meta.region_name}::/restapis/{api['id']}/stages/{stage['stageName']}"
+                    for stage in stages['item']
+                    if not stage['methodSettings']
+                ])
+                non_compliant_resources.extend([
+                    f"arn:aws:apigateway:{v1_client.meta.region_name}::/restapis/{api['id']}/stages/{stage['stageName']}"
+                    for stage in stages['item']
+                    for _,settings in stage['methodSettings'].items()
+                    if not (settings['loggingLevel'] == "INFO"
+                            or settings['loggingLevel'] == "ERROR")
+                ])
```

Comment:
settings['loggingLevel']이 존재하지 않아 KeyError가 발생하는 경우가 있습니다.
```suggestion
                non_compliant_resources.extend([
                    f"arn:aws:apigateway:{v1_client.meta.region_name}::/restapis/{api['id']}/stages/{stage['stageName']}"
                    for stage in stages['item']
                    for _,settings in stage['methodSettings'].items()
                    if not 'loggingLevel' in settings
                    or settings['loggingLevel'] == 'OFF'
                ])
```